diff --git a/lib/spack/spack/config.py b/lib/spack/spack/config.py
index 1af333e7..1df62d42 100644
--- a/lib/spack/spack/config.py
+++ b/lib/spack/spack/config.py
@@ -759,13 +759,24 @@ def override(
         assert scope is overrides
 
 
-def _add_platform_scope(cfg: Configuration, name: str, path: str, writable: bool = True) -> None:
-    """Add a platform-specific subdirectory for the current platform."""
-    platform = spack.platforms.host().name
-    scope = DirectoryConfigScope(
-        f"{name}/{platform}", os.path.join(path, platform), writable=writable
-    )
-    cfg.push_scope(scope)
+def _add_platform_scope(cfg: Configuration, name: str, path: str, writable: bool =True) -> None:
+    """Add subdirectories for the current platform, os, and target."""
+    host_platform = spack.platforms.host()
+    platform = host_platform.name
+    oss = str(host_platform.operating_system("frontend"))
+    host_target = str(host_platform.target("frontend"))
+
+    scope_name = os.path.join(name, platform)
+    scope_path = os.path.join(path, platform)
+    cfg.push_scope(DirectoryConfigScope(scope_name, scope_path, writable=writable))
+
+    scope_name = os.path.join(scope_name, oss)
+    scope_path = os.path.join(scope_path, oss)
+    cfg.push_scope(DirectoryConfigScope(scope_name, scope_path, writable=writable))
+
+    scope_name = os.path.join(scope_name, host_target)
+    scope_path = os.path.join(scope_path, host_target)
+    cfg.push_scope(DirectoryConfigScope(scope_name, scope_path, writable=writable))
 
 
 def config_paths_from_entry_points() -> List[Tuple[str, str]]:
